### PR TITLE
Replaced logging methods with lazy function based versions

### PR DESCRIPTION
**Pull request by sksamuel**

### CODE REVIEW

File: kotest-framework/kotest-framework-discovery/src/jvmMain/kotlin/io/kotest/framework/discovery/Discovery.kt [hunk 1]
```diff
@@ -167,10 +168,10 @@ class Discovery(private val discoveryExtensions: List<DiscoveryExtension> = empt
     */
    private fun scanUris(): List<KClass<out Spec>> {
       val uptime = ManagementFactory.getRuntimeMXBean().uptime
-      log("Discovery: Starting test discovery scan... [uptime=$uptime]")
+      log { "Discovery: Starting test discovery scan... [uptime=$uptime]" }
       val start = System.currentTimeMillis()
       val duration = System.currentTimeMillis() - start
-      log("Discovery: Test discovery completed in ${duration}ms")
+      log { "Discovery: Test discovery completed in ${duration}ms" }
```
Comment (lwasyl) [inline kotest-framework/kotest-framework-discovery/src/jvmMain/kotlin/io/kotest/framework/discovery/Discovery.kt]: `uptime` and `start/duration` could probably go inside the lambdas as well?

File: kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/KotestEngine.kt [hunk 2]
```diff
@@ -112,35 +112,35 @@ class KotestEngine(private val config: KotestEngineConfig) {
...
+         log { "KotestEngine: Script execution completed" }
       }
 
       // spec classes are ordered using an instance of SpecExecutionOrder
-      log("KotestEngine: Sorting specs by ${configuration.specExecutionOrder}")
+      log { "KotestEngine: Sorting specs by ${configuration.specExecutionOrder}" }
       val ordered = plan.classes.sort(configuration.specExecutionOrder)
 
-      log("KotestEngine: Will use spec executor ${configuration.specExecutionOrder}")
+      log { "KotestEngine: Will use spec executor ${configuration.specExecutionOrder}" }
```
Comment (lwasyl) [inline kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/KotestEngine.kt]: Was this originally supposed to be `use spec executor $executor`? order is logged above already :)

**Review by lwasyl — APPROVED** [review]

File: kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/KotestEngine.kt (hunk 2 above)
Comment (sksamuel) [reply kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/KotestEngine.kt]: thanks fixed

File: kotest-framework/kotest-framework-discovery/src/jvmMain/kotlin/io/kotest/framework/discovery/Discovery.kt (hunk 1 above)
Comment (sksamuel) [reply kotest-framework/kotest-framework-discovery/src/jvmMain/kotlin/io/kotest/framework/discovery/Discovery.kt]: thanks fixed

**Pull request merged**